src/frontend/Semantic_error.ml

```diff
@@ -80,12 +80,16 @@ module TypeError = struct
           (Pretty_printing.pretty_print_assignmentoperator assignop)
           UnsizedType.pp lt UnsizedType.pp rt
     | IllTypedTernaryIf (ut1, ut2, ut3) ->
-        Fmt.pf ppf
-          "Ill-typed arguments supplied to ? : operator. Available \
-           signatures: %s\n\
-           Instead supplied arguments of incompatible type: %a, %a, %a."
-          (Stan_math_signatures.pretty_print_math_sigs "if_else")
-          UnsizedType.pp ut1 UnsizedType.pp ut2 UnsizedType.pp ut3
+        if ut1 = UnsizedType.UInt then
```

Comment:
This should just be part of the pattern match on line 82 I think. You'll end up with two lines,

```ocaml
    | IllTypedTernaryIf (UInt, ut2, ut3) -> ...
    | IllTypedTernaryIf (ut1, ut2, ut3) ->...
```